File: beekeeping-app/src/app/_models/queen.ts
import { Colors } from "./beehive";

export class Queen {
    id: number;
    breed: Breed;
    state: QueenState;
    hatchingDate: Date;
    markingColor: Colors;
    isFertilized: boolean;
    broodStartDate: Date;
    farmId: number;
}

export enum QueenState {
    Cell = 1,
    LvingInBeehive = 2,
   // PriduodamaŠeimai = 4, //kol kas nenaudojama
    Isolated = 8,
    Selled = 16,
    Swarmed = 32,
    Dead = 64
}

export const QueenState2LabelMapping: Record<QueenState, string> = {
    [QueenState.Cell]: "Lopšys",
    [QueenState.LvingInBeehive]: "Gyvena avilyje",
//    [QueenState.PriduodamaŠeimai]: "Priduodama šeimai",
    [QueenState.Isolated]: "Izoliuota narvelyje",
    [QueenState.Selled]: "Parduota",
    [QueenState.Swarmed]: "Išsispietusi",
    [QueenState.Dead]: "Numirusi"
};

export enum Breed {
    Bakfasto, Karnika, Mišrūnė, Nežinoma
}

export const Breed2LabelMapping: Record<Breed, string> = {
    [Breed.Bakfasto]: "Bakfasto",
    [Breed.Karnika]: "Karnika",
    [Breed.Mišrūnė]: "Mišrūnė",
    [Breed.Nežinoma]: "Nežinoma"
};

File: beekeeping-api/BeekeepingApi/Models/Queen.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace BeekeepingApi.Models
{
    public enum Breed
    {
        Bakfasto, Karnika, Mišrūnė, Nežinoma
    }

    public enum QueenState
    {
        Lopšys = 1, //Cell
        GyvenaAvilyje = 2,
        PriduodamaŠeimai = 4,
        IzoliuotaNarvelyje = 8,
        Parduota = 16,
        Išsispietusi = 32,
        Numirusi = 64
    }

    public class Queen
    {
        [Key]
        public long Id { get; set; }

        public Breed Breed { get; set; }

        public DateTime? HatchingDate { get; set; }

        public Colors? MarkingColor { get; set; }

        public bool IsFertilized { get; set; }

        public DateTime? BroodStartDate { get; set; }

        public QueenState State { get; set; }

        public long FarmId { get; set; }

        [ForeignKey("FarmId")]
        public Farm Farm { get; set; }

        public virtual ICollection<BeefamilyQueen> BeeFamilyQueens { get; set; }
        public virtual ICollection<QueensRaising> QueensRaisings { get; set; }
    }
}